docs/partitions.md

```diff
@@ -0,0 +1,92 @@
+# Partitions
+
+In the original implementation of ringpop, if a cluster is split to multiple
+partitions, nodes in each partition declare each other as faulty, and don't
+communicate. Ringpop implemented support for merging the partitions, which we
+call `healing`.
+
+## Introdction -- basic algorithm
+
+In order for two partitions to heal, the algorithm does the following,
+periodically (number of times per time unit is upper bounded):
```

Comment:
I find it hard to interpret this sentence (I know what you mean but I think that's because I know the algorithm)